net/net.go

```diff
@@ -1022,37 +1032,22 @@ func (n *net) startPulling() {
 	}
 }
 
-// getOwnLoad returns the log owned by the host under the given thread.
-func (n *net) getOwnLog(id thread.ID) (info thread.LogInfo, err error) {
-	logs, err := n.store.LogsWithKeys(id)
-	if err != nil {
-		return
-	}
-	for _, lid := range logs {
-		sk, err := n.store.PrivKey(id, lid)
-		if err != nil {
-			return info, err
-		}
-		if sk != nil {
-			return n.store.GetLog(id, lid)
-		}
-	}
-	return info, nil
-}
-
-// getOrCreateOwnLoad returns the log owned by the host under the given thread.
-// If no log exists, a new one is created under the given thread.
+// getOrCreateOwnLoad returns the first log 'owned' by the host under the given thread.
+// If no log exists, a new one is created under the given thread and returned.
+// This is a strict 'owership' check, vs returning the first directly 'managed' log.
 func (n *net) getOrCreateOwnLog(id thread.ID) (info thread.LogInfo, err error) {
-	info, err = n.getOwnLog(id)
+	lgs, err := n.store.GetOwnLogs(id)
 	if err != nil {
 		return info, err
 	}
-	if info.PubKey != nil {
-		return
+	for _, info = range lgs {
+		if info.PrivKey != nil {
+			return info, nil
```

Comment:
Seems like we need to ensure somewhere (maybe this function _is_ that assurance?) that a thread does not have multiple private keys (I'm not if we're missing out on some use case_. Above I mentioned something about having multiple private keys, but maybe not relevant considering this...

Reply:
Yeh, in theory, it shouldn't be possible to get into a state where there are multiple private keys, because as you've identified, this function essentially checks that. I'm going to double check, but I think this will _always_ be called any time a 'managed' log is being created, so this check should be sufficient.

Reply:
Cool, sounds good! I don't remember, but you might be able to call `AddThread` multiple times for the same thread... and both times (or just the second time) supply your _own_ private log key.

Reply:
Looked... of course, you can't add the same thread twice. So, as it stands now, a thread can only have _one direct/managed log_. That makes sense actually, since in the multitenant setup (single writer), multiple identities will just use the same log.

Reply:
Ah yes of course. But what about the situation where we have multiple remote clients treating a single remote peer as essentially a relay node? In that case, if one client is already using the remote to interact with a thread, the second client can't also use it. The 'single writer' setup isn't going to cover that use case, is it?

Reply:
(That msg is more about thinking through how we might support this use-case, which is actually what motivated this PR/Issue in the first place)

Reply:
Note that the answer might also be, we're not going to support that use-case directly, at this time...?

Reply:
Hmm, yeah good point. I guess we should just allow it via `CreateThread` / `AddThread`. I think you'd have to ensure info isn't overwritten in `logstore.AddThread` when it's called multiple times for the same thread (probably okay, but just occurred to me as something to look at). Also, `ensureUnique` might have to get smarter, taking the logkey into account.

Reply:
Phrew, ok, those points are the same conclusions I came to as well! That's fine then. I think totally doable in this PR update tomorrow.